test/invariant/handlers/TransferHandler.sol

```diff
@@ -9,29 +9,64 @@ import { PSM3 } from "src/PSM3.sol";
 
 contract TransferHandler is HandlerBase {
 
+    MockERC20[3] public assets;
+
     uint256 public transferCount;
 
+    mapping(address asset => uint256) public transfersIn;
+
     constructor(
         PSM3      psm_,
         MockERC20 asset0,
         MockERC20 asset1,
         MockERC20 asset2
-    ) HandlerBase(psm_, asset0, asset1, asset2) {}
+    ) HandlerBase(psm_) {
+        assets[0] = asset0;
+        assets[1] = asset1;
+        assets[2] = asset2;
+    }
+
+    function _getAsset(uint256 indexSeed) internal view returns (MockERC20) {
+        return assets[indexSeed % assets.length];
+    }
 
     function transfer(uint256 assetSeed, string memory senderSeed, uint256 amount) external {
-        MockERC20 asset  = _getAsset(assetSeed);
+        // 1. Setup and bounds
+
+        MockERC20 asset = _getAsset(assetSeed);
         address   sender = makeAddr(senderSeed);
 
+        // 2. Cache starting state
+        uint256 startingConversion = psm.convertToShares(1e18);
+        uint256 startingValue      = psm.totalAssets();
+
         // Bounding to 10 million here because 1 trillion introduces unrealistic conditions with
         // large rounding errors. Would rather keep tolerances smaller with a lower upper bound
         // on transfer amounts.
         amount = _bound(amount, 1, 10_000_000 * 10 ** asset.decimals());
 
+        // 3. Perform action against protocol
         asset.mint(sender, amount);
-
         vm.prank(sender);
         asset.transfer(address(psm), amount);
 
+        // 4. Update ghost variable(s)
+        transfersIn[address(asset)] += amount;
+
+        // 5. Perform action-specific assertions
+        assertGe(
+            psm.convertToAssetValue(1e18) + 1,
```

Comment:
Same here.

Reply:
Updated to use `convertToAssetValue` for both